internal/secret/string.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package secret
+
+import (
+	"fmt"
+
+	"github.com/goccy/go-yaml"
+)
+
+type Secret[T any] interface {
```

Comment:
I did not use this interface currently, but I figure if ever we are going to need another type of secret data I decided to make a generic interface. We may decide to remove it later. Currently, it's not causing any harm.